main.go

```diff
@@ -246,11 +332,17 @@ func (m *Miner) hashratePrinter() {
 }
 
 // resultLoop takes in the result and passes to the proper channels for receiving.
-func (m *Miner) resultLoop() error {
+func (m *Miner) resultLoop(config util.Config) {
 	for {
 		select {
 		case header := <-m.resultCh:
 			order, err := m.GetDifficultyOrder(header)
```

Comment:
Improved error handling when this function returns with an error. Previously, it would have still tried to send the mined header. Now it will stop execution on this header and wait for a new one to arrive in the channel.